### PR TITLE
Add az diagnostic log collection

### PR DESCRIPTION
**Pull request by mpagot**
Add some code to collect the diagnostic logs from any VM in the resource group. Add it to some of the post fail hook of the ipaddr2 test.

- Related ticket: https://jira.suse.com/browse/TEAM-8721

# Verification run: 
sle-15-SP5-SapCloud-Azure-Payg-x86_64-Buildmpagot_VR-ipaddr2_azure_test@64bit 
- http://openqaworker15.qa.suse.cz/tests/287030  with a simulated failure --> http://openqaworker15.qa.suse.cz/tests/287030#step/configure/28 the log
- http://openqaworker15.qa.suse.cz/tests/286951  with a simulated failure and without the setting

### CODE REVIEW

**Comment by badboywj** [comment]
LGTM, thanks

File: lib/sles4sap/ipaddr2.pm [hunk 1]
```diff
@@ -236,6 +242,19 @@ sub ipaddr2_azure_deployment {
         frontend_ip => $lb_fe,
         name => $lb . "_rule",
         port => '80');
+
+    if ($args{diagnostic} eq 1) {
```
Comment (BillAnastasiadis) [inline lib/sles4sap/ipaddr2.pm:246]: Can't it also be 
`if ($args{diagnostic})`
?

**Review by alvarocarvajald — APPROVED** [review]
LGTM

File: lib/sles4sap/ipaddr2.pm (hunk 1 above)
Comment (alvarocarvajald) [reply lib/sles4sap/ipaddr2.pm:246]: Agree with Bill. `eq 1` will:

- Cast `$args{diagnostic}` into a string.
- Cast `1` into a string.
- Do a string equal check.

So, `if ($args{diagnostic}` will already do a conversion to boolean and evaluate it which is a bit better.

**Pull request merged**